sql/core/src/main/scala/org/apache/spark/sql/execution/joins/MergeAsOfJoinExec.scala

```diff
@@ -206,19 +204,30 @@ private class MergeAsOfIterator(
     // The current groups should be matching and the group should not be empty.
     assert(currRightIter.hasNext)
     var rHead = currRightIter.next()
-    var rPrev = rHead.copy()
```

Comment:
I think you probably want to keep the copy here to maintain the invariant "rPrev" is always copied